core/jobs/batch_jobs/exploration_migration_jobs.py

```diff
@@ -0,0 +1,325 @@
+
+        Returns:
+            Result((str, Exploration), (str, Exception)). Result
+            containing tuple that consists of exploration ID and either
+            exploration object or Exception. Exploration object is
+            returned when the migration was successful and Exception
+            is returned otherwise.
+        """
+        try:
+            exploration = exp_fetchers.get_exploration_from_model(exp_model)
+            exploration.validate()
+        except Exception as e:
+            logging.exception(e)
+            return result.Err((exp_id, e))
+
+        return result.Ok((exp_id, exploration))
+
+    @staticmethod
+    def _update_exploration(
+        exp_model: exp_models.ExplorationModel,
+        migrated_exploration: exp_domain.Exploration,
+        exploration_changes: Sequence[exp_domain.ExplorationChange]
+    ) -> Sequence[base_models.BaseModel]:
+        """Generates newly updated exploration models.
+
+        Args:
+            exp_model: ExplorationModel. The exploration which
+                should be updated.
+            migrated_exploration: Exploration. The migrated exploration
+                domain object.
+            exploration_changes: sequence(ExplorationChange). The
+                exploration changes to apply.
+
+        Returns:
+            sequence(BaseModel). Sequence of models which should be put into
+            the datastore.
+        """
+        updated_exploration_model = (
+            exp_services.populate_exploration_model_fields(
+                exp_model, migrated_exploration))
+        commit_message = (
+            'Update exploration content schema version to %d.'
+        ) % (feconf.CURRENT_EXP_SCHEMA_VERSION)
+        change_dicts = [change.to_dict() for change in exploration_changes]
+        with datastore_services.get_ndb_context():
+            models_to_put = updated_exploration_model.compute_models_to_commit(
+                feconf.MIGRATION_BOT_USERNAME,
+                feconf.COMMIT_TYPE_EDIT,
+                commit_message,
+                change_dicts,
+                additional_models={}
+            ).values()
+        datastore_services.update_timestamps_multi(list(models_to_put))
+        return models_to_put
+
+    @staticmethod
+    def _update_exploration_summary(
+        migrated_exploration: exp_domain.Exploration,
+        exploration_summary_model: exp_models.ExpSummaryModel
+    ) -> exp_models.ExpSummaryModel:
+        """Generates newly updated exploration summary model.
+
+        Args:
+            migrated_exploration: Exploration. The migrated exploration
+                domain object.
+            exploration_summary_model: ExpSummaryModel. The exploration
+                summary model to update.
+
+        Returns:
+            ExpSummaryModel. The updated exploration summary model to put into
+            the datastore.
+        """
+        with datastore_services.get_ndb_context():
```

Comment:
Refactor the summary creation so that this is not needed. This should be only used when we create models and not for allowing any get/put/delete operation.

Reply:
Without this, it is throwing error
```
google.cloud.ndb.exceptions.ContextError: No current context. NDB calls must be made in context established by google.cloud.ndb.Client.context. [while running 'Generate exploration summary models to put']
```

Reply:
In that case you need to modify the code so that we pass the `ExplorationRightsModel` and `ExpSummaryModel` through the Beam job

Reply:
This is not working. Without `with datastore_services.get_ndb_context():`, the get of `ExplorationRightsModel ` and `ExpSummaryModel` is not working. 
Or do you think there is any other reason for this error?

Reply:
Yep, it is because in `compute_summary_of_exploration` you use NDB operations.